### PR TITLE
[KYUUBI #383] Support beeline --hivevar --hiveconf

### LINKED ISSUE #383: Can not access hivevar set by `beeline --hivevar`
**Describe the bug**
Can not access hivevar set by `beeline --hivevar` 

**To Reproduce**
1. Start a session using `beeline --hivevar abc=xyz`
2. Run `select ${abc}` throw exception
3. Run `select ${spark.abc}` got `xyz`

**Expected behavior**
Respect beeline hive behavior, access `--hivevar abc` by `${abc}` or `${hivevar:abc}`.

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by pan3793**
pan3793 [Closes #394](https://github.com/yaooqinn/kyuubi/pull/394)    [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

close #383

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 1]
```diff
@@ -54,8 +54,12 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
     try {
       val sparkSession = spark.newSession()
       conf.foreach {
-        case (HIVE_VAR_PREFIX(key), value) => setModifiableConfig(sparkSession, key, value)
-        case (HIVE_CONF_PREFIX(key), value) => setModifiableConfig(sparkSession, key, value)
+        case (HIVE_VAR_PREFIX(key), value) if key.startsWith(SPARK_PREFIX) =>
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:57]: can we remove HIVE_VAR_PREFIX and HIVE_VAR_PREFIX, just use `if else` like KyuubiSessionImpl?

Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:57]: I found some negative cases that these regexes that didn't match well, so I switch to match case by case

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:57]: OK, I will change it

**Comment by yaooqinn** [comment]
I suggest we add a test case at the server side too

**Comment by yaooqinn** [comment]
can you rebase master to retrigger CI? the CI is flaky now

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**